### PR TITLE
Update dependencies and apply select linter suggestions

### PR DESCRIPTION
**Pull request by Rupan**
I went through the exercise of building TagMo from source code.  As part of this process I did a quick pass over the build artifacts and discovered that many were significantly out of date.  JCenter, for example, is deprecated and should be replaced by Maven Central.  Also, a number of the XML artifacts use attributes which are not supported until Android SDK 23 so I updated the minimum SDK version.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -3,10 +3,8 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.hiddenramblings.tagmo">
 
-    <uses-permission android:name="android.permission.BIND_NFC_SERVICE"/>
     <uses-permission android:name="android.permission.NFC"/>
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
-    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
```
Comment (burndive) [inline app/src/main/AndroidManifest.xml:9]: Will removing WRITE_EXTERNAL_STORAGE here prevent the app from saving scanned bin files to external storage?

Comment (Rupan) [reply app/src/main/AndroidManifest.xml:9]: Thanks for pointing this out; after double checking, it turns out that the removal was based on a misunderstanding of the documentation:

https://developer.android.com/reference/android/Manifest.permission#WRITE_EXTERNAL_STORAGE
https://developer.android.com/reference/android/os/Environment#getExternalStorageDirectory()

This API will break from Android Q onward and the app needs an update to handle future devices.  But the devices with API level 23 and above will break if this is removed, so it should be kept.

**Comment by Rupan** [comment]
Are there any further concerns for this PR? Can it be merged?

**Comment by AbandonedCart** [comment]
It seems a bit "hacky" to update everything and then lower the API to avoid scoped storage. If you honestly feel there is a reason other than having to write new code, then you may want to look into https://developer.android.com/reference/android/R.attr#preserveLegacyExternalStorage so that Android does not come along and override the now deprecated flag in a future update.

**Comment by Rupan** [comment]
@AbandonedCart the reason for rolling back to API level 29 was specifically to avoid writing new code.  This specific PR is about updating dependencies and fixing linter warnings, with an aim towards a simple code review.  Adding brand new features should happen in a separate pull request.

**Comment by AbandonedCart** [comment]
Seeing as it is over a year since the last merge, I don't see it happening.

**Comment by AbandonedCart** [comment]
Superseded by ScopedStorage in to https://github.com/HiddenRamblings/TagMo/tree/experimental release

**Pull request closed without merging**